### PR TITLE
Project 4 Chat bot collaborator Li Lindblom

### PR DESCRIPTION
**Pull request by amandatange**

### CODE REVIEW

File: code/index.html [hunk 1]
```diff
@@ -3,24 +3,27 @@
...
     <meta charset="UTF-8" />
-    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0"/>
     <link rel="stylesheet" href="./style.css" />
     <link
       href="https://fonts.googleapis.com/css2?family=Montserrat:ital,wght@0,100;0,200;0,300;0,400;0,500;0,600;0,700;0,800;0,900;1,100;1,200;1,300;1,400;1,500;1,600;1,700;1,800;1,900&display=swap"
       rel="stylesheet" />
-    <title>Chat Bot</title>
+    <title>Mood Bot</title>
+    <link rel='stylesheet' href='https://www.cnap.fr/sites/faune/styles/faune-fontes.css'>
```
Comment (michaelchangdk) [inline code/index.html:12]: I'm really curious what this is!

File: code/script.js [hunk 2]
```diff
@@ -1,10 +1,203 @@
...
+const form = document.getElementById("name-form");
+
+// Global variables
+let numberOfQuestionsAsked = 0; //counter for number of questions
+
+let allKnowingObj = {}; //To save all the answers and make them available in the global scope
+
+const ageAlternatives = ["0-29", "30-59", "60+"]
+
+const smileyHTMLText = ["&#128520", "&#128525", "&#128549", "&#128556", "&#128513"];
```
Comment (michaelchangdk) [inline code/script.js:15]: Wow this is so smart! I didn't even think about the fact that you could store the options as an array. Awesome!

File: code/script.js [hunk 3]
```diff
@@ -1,10 +1,203 @@
...
+//Choices for the dropdown menu
+const weatherArray = ["Sunny", "Cloudy", "Rainy", "Snowy", "Stormy", "Windy", "Hail"];
+
+//Gradients displaying if last answer is yes
+const blueish = "linear-gradient(90deg, rgba(10,41,84,1) 0%, rgba(10,63,64,1) 30%, rgba(19,74,92,1) 53%, rgba(26,49,112,1) 72%, rgba(24,21,91,1) 88%)";
+const greenish = "linear-gradient(90deg, rgba(18,8,80,1) 0%, rgba(11,52,9,1) 30%, rgba(50,92,19,1) 53%, rgba(26,112,49,1) 72%, rgba(21,91,73,1) 88%)";
+const redish = "linear-gradient(90deg, rgba(84,10,10,1) 0%, rgba(64,25,10,1) 30%, rgba(92,76,19,1) 53%, rgba(96,112,26,1) 72%, rgba(91,64,21,1) 88%)";
+const yellowish = "linear-gradient(90deg, rgba(51,48,11,1) 0%, rgba(75,64,14,1) 30%, rgba(90,92,19,1) 53%, rgba(131,158,55,1) 72%, rgba(176,190,34,1) 88%)";
+const purpleish = "linear-gradient(90deg, rgba(48,11,51,1) 0%, rgba(21,14,75,1) 30%, rgba(75,19,92,1) 53%, rgba(78,26,70,1) 72%, rgba(131,27,49,1) 88%)";
+const backgroundGradients = [blueish, redish, greenish, yellowish, purpleish];
```
Comment (michaelchangdk) [inline code/script.js:30]: Super cool! Again - awesome concept and really interesting using the gradients. Definitely something I will be borrowing/need to learn more about!

File: code/script.js [hunk 4]
```diff
@@ -1,10 +1,203 @@
...
+  
+  numberOfQuestionsAsked++; //adds 1 to the counter
+
+  if (numberOfQuestionsAsked < 6) {
+    if (numberOfQuestionsAsked === 1) {
+      showMessage(`My name is ${input.value}`, "user");
+      //saves the user's name in the object
+      allKnowingObj["name"] = input.value;
+      setTimeout(() => {
+        showMessage(`Hello ${allKnowingObj["name"]}! So you're wondering what your mood is? Let me help!
```
Comment (michaelchangdk) [inline code/script.js:57]: Ooh clever with the allKnowingObj!!!

File: code/script.js [hunk 5]
```diff
@@ -1,10 +1,203 @@
...
+      } else {
+        setTimeout(() => {
+          changeBackground("black"); //if user is not happy they get the colour black
+        }, 500);
+      };
+    }, 500);
+    inputWrapper.innerHTML = "";
+  };
+  input.value = '';
+};
```
Comment (michaelchangdk) [inline code/script.js:90]: The whole way this is built is so elegant. I feel like somebody more qualified than me should be reviewing this!

File: code/script.js [hunk 6]
```diff
@@ -1,10 +1,203 @@
...
+    } else if (numberOfQuestionsAsked === 5) {
+      showMessage(allKnowingObj["weather-choice"], "user");
+      setTimeout(questionHappy, 500); //Calls last question
+    };
+  } else { //Triggered if there have been 6 questions asked
+    setTimeout(() => {
+      showMessage("Thank you for your input! Here is your mood colour!", "bot");
+      if (allKnowingObj["happy"].includes("yes") || allKnowingObj["happy"].includes("oui")) { //if user is happy with the service they get a random colour form the gradient array
+        setTimeout(() => {
+          changeBackground(backgroundGradients[randomIndex()]);
```
Comment (michaelchangdk) [inline code/script.js:79]: Wait so the color we get is random??? 😮

File: code/script.js [hunk 7]
```diff
@@ -1,10 +1,203 @@
...
+  inputWrapper.innerHTML = "";
+
+  showMessage(`Alright, what's your energy level?`,"bot"); 
+  
+  //re-appends the form and change of input to range 
+  inputWrapper.appendChild(form);
+  form.appendChild(min_);
+  form.appendChild(input);
+  form.appendChild(max_);
+  form.appendChild(sendButton);
```
Comment (michaelchangdk) [inline code/script.js:120]: Wow this is amazing. I need to read up about appendChild!

File: code/style.css [hunk 8]
```diff
@@ -134,7 +138,57 @@ button {
...
+  -webkit-appearance: none;  /* Override default CSS styles */
+  appearance: none;
+  width: 100%; /* Full-width */
+  height: 25px; /* Specified height */
+  background: #d3d3d3; /* Grey background */
+  outline: none; /* Remove outline */
+  opacity: 0.7; /* Set transparency (for mouse-over effects on hover) */
+  -webkit-transition: .2s; /* 0.2 seconds transition on hover */
+  transition: opacity .2s;
+}
```
Comment (michaelchangdk) [inline code/style.css:169]: Styled sliders have killed me before.. I'm definitely stealing this snippet 😉

**Review by michaelchangdk — COMMENTED** [review]
This was built so differently than how we approached our project. I'm still trying to wrap my head around a lot of what's happening in your code. I feel like the code coaches should be reviewing your project right now, not me! Super impressed and a lot to learn / take in. Really enjoyed reading through this. ⭐ ⭐ ⭐ ⭐ ⭐  5 stars!

File: code/index.html [hunk 9]
```diff
@@ -3,24 +3,27 @@
...
   </head>
 
   <body>
-    <h1>Welcome to xxx.</h1>
-    <h2>Ask about xxxx.</h2>
+    <header> 
+    <h1>Welcome to Mood Bot</h1>
+    <h2>Find out what colour your mood is!</h2>
```
Comment (Kras053) [inline code/index.html:18]: this made me think of those mood rings you can buy that changes color, original idea for a chat bot!

File: code/script.js [hunk 10]
```diff
@@ -1,10 +1,203 @@
...
+let numberOfQuestionsAsked = 0; //counter for number of questions
+
+let allKnowingObj = {}; //To save all the answers and make them available in the global scope
+
+const ageAlternatives = ["0-29", "30-59", "60+"]
+
+const smileyHTMLText = ["&#128520", "&#128525", "&#128549", "&#128556", "&#128513"];
+
+//min and max values for the energy range slider
+const min_ = document.createElement("p");
```
Comment (Kras053) [inline code/script.js:18]: nice feature with the energy range slider!

File: code/script.js (hunk 3 above)
Comment (Kras053) [reply code/script.js:30]: +1 : D

File: code/script.js [hunk 11]
```diff
@@ -1,10 +1,203 @@
...
+  } else { //Triggered if there have been 6 questions asked
+    setTimeout(() => {
+      showMessage("Thank you for your input! Here is your mood colour!", "bot");
+      if (allKnowingObj["happy"].includes("yes") || allKnowingObj["happy"].includes("oui")) { //if user is happy with the service they get a random colour form the gradient array
+        setTimeout(() => {
+          changeBackground(backgroundGradients[randomIndex()]);
+        }, 1500);
+      } else {
+        setTimeout(() => {
+          changeBackground("black"); //if user is not happy they get the colour black
```
Comment (Kras053) [inline code/script.js:83]: Oh my guess was right haha - fun to get to see the idea behind the color choices!

**Review by Kras053 — COMMENTED** [review]
Well done with the chat-bot-project Amanda and Li! It was really interesting to read your code, it felt ambitious and well structured, and an original idea! 
I feel inspired to learn more about loops, gradients and appendChild and wish I could give more feedback on that. 
Great job!

**Pull request closed without merging**